test_integration/test_ee_short_region_slop.py

```diff
@@ -0,0 +1,300 @@
+        min_runtime = min(all_runtime)
+        max_runtime = max(all_runtime)
+        delta_runtime = max_runtime - min_runtime
+        min_runtime -= 0.05 * delta_runtime
+        max_runtime += 0.05 * delta_runtime
+
+        plt.figure(figsize=(11,11))
+        plt.subplot(2, 2, 1)
+        plt.title('Scaling region package-energy')
+        plot_data(cols, scaling_data_fixed, 'duration', 'package-energy')
+        plot_data(cols, scaling_data_dynamic, 'duration', 'package-energy')
+        plt.ylim(min_energy, max_energy)
+        plt.legend(('fixed', 'dynamic'))
+
+        plt.subplot(2, 2, 2)
+        plt.title('Timed region package-energy')
+        plot_data(cols, timed_data_fixed, 'duration', 'package-energy')
+        plot_data(cols, timed_data_dynamic, 'duration', 'package-energy')
+        plt.ylim(min_energy, max_energy)
+        plt.legend(('fixed', 'dynamic'))
+
+        plt.subplot(2, 2, 3)
+        plt.title('Scaling region runtime')
+        plot_data(cols, scaling_data_fixed, 'duration', 'runtime')
+        plot_data(cols, scaling_data_dynamic, 'duration', 'runtime')
+        plt.ylim(min_runtime, max_runtime)
+        plt.legend(('fixed', 'dynamic'))
+
+        plt.subplot(2, 2, 4)
+        plt.title('Timed region runtime')
+        plot_data(cols, timed_data_fixed, 'duration', 'runtime')
+        plot_data(cols, timed_data_dynamic, 'duration', 'runtime')
+        plt.ylim(min_runtime, max_runtime)
+        plt.legend(('fixed', 'dynamic'))
+        plt.savefig(self._image_path)
+
+def extract_data(report):
+    cols = [('count', ''),
+            ('package-energy', 'joules'),
+            ('requested-online-frequency', ''),
+            ('power', 'watts'),
+            ('runtime', 'sec'),
+            ('frequency', 'Hz')]
+    scaling_data = extract_data_region(report, cols, 'scaling')
+    timed_data = extract_data_region(report, cols, 'timed')
+    cols.append(('duration', 'sec'))
+    dur = tuple(rt / ct for ct, rt in zip(scaling_data[column_idx(cols, 'count')],
+                                          scaling_data[column_idx(cols, 'runtime')]))
+    scaling_data.append(dur)
+    dur = tuple(rt / ct for ct, rt in zip(timed_data[column_idx(cols, 'count')],
+                                          timed_data[column_idx(cols, 'runtime')]))
+    timed_data.append(dur)
+    return cols, scaling_data, timed_data
+
+def column_idx(cols, name):
+    return zip(*cols)[0].index(name)
+
+def plot_data(cols, data, xaxis, yaxis):
+    xaxis_idx = column_idx(cols, xaxis)
+    yaxis_idx = column_idx(cols, yaxis)
+    plt.semilogx(data[xaxis_idx], data[yaxis_idx], 'x')
+    units = cols[xaxis_idx][1]
+    if units:
+        plt.xlabel('{} ({})'.format(xaxis, units))
+    else:
+        plt.xlabel(xaxis)
+    units = cols[yaxis_idx][1]
+    if units:
+        plt.ylabel('{} ({})'.format(yaxis, units))
+    else:
+        plt.ylabel(yaxis)
+
+def extract_data_region(report, cols, key):
+    host = report.host_names()[0]
```

Comment:
could pass in host as a parameter?

Reply:
Could do, and if we promoted this implementation to a separate object that would make sense.  In this case the test is only run on one host.